### ISSUE TITLE
gradient descent and performance issues with Tf 2.2

### ISSUE THREAD

**Issue by lordfiftyfive** [tensorflow/probability#948]
I am running an algorithm with tensorflow 2.2 and tensorflow-probability 0.10 using a gaussian process layer which is based on the example provided in google colab. 

I am getting the following warnings which seem to be impacting my performance further. Before the upgrade I could get near 0 loss with 270 epochs. Now it has stopped improving altogether after a few hundred epochs and gets permanently stuck at 34. I believe that the problem may have something to do with this warning 
`WARNING:tensorflow:Gradients do not exist for variables ['variational_gaussian_process_6/observation_noise_variance:0'] when minimizing the loss.`

I suspect that while the VGP distribution was made tape safe the VGP layer still doesnt appear to be tape safe. I am wondering if this is true and if so how I can get around this problem
